src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -46,13 +56,27 @@ public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player,
         player.saveComponent(thirst);
     }
 
+    /**
+     * Defines what to do when an entity is removed.
+     *
+     * @param event  the event corresponding to the removal of an entity
+     * @param entity the entity being removed
+     * @param thirst the thirst component associated with the entity
+     */
     @ReceiveEvent
     public void beforeRemoval(BeforeDeactivateComponent event, EntityRef entity, ThirstComponent thirst) {
         thirst.lastCalculatedWater = ThirstUtils.getThirstForEntity(entity);
         thirst.lastCalculationTime = time.getGameTimeInMs();
         entity.saveComponent(thirst);
     }
 
+    /**
+     * Defines what happens when a drink is consumed.
```

Comment:
Same as above: should state something like "applies the drink's filling to the instigator of the drinking action (ActivationEvent)"